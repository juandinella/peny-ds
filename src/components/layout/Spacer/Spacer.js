import React from 'react'
import PropTypes from 'prop-types'
import styles from './Spacer.styles'
import Horizontal from './components/Horizontal/Horizontal'
import Vertical from './components/Vertical/Vertical'
import { options } from './constants'

/**
 * Provides empty space.
 */

const Spacer = ({ id, mSize, size, isVisible }) => (
  <div
    css={styles}
    id={id}
    data-testid={id}
    size={size}
    mSize={mSize}
    isVisible={isVisible}
  />
)

Spacer.displayName = 'Spacer'

Spacer.propTypes = {
  /** Differents sizes of the spacer */
  size: PropTypes.oneOf(options.sizes),
  /* Size for mobile */
  mSize: PropTypes.oneOf(options.sizes),
  /** This Boolean attribute add border to see it on Storybook */
  isVisible: PropTypes.bool,
  /** Spacer's id */
  id: PropTypes.string,
}

Spacer.defaultProps = {
  size: '4',
  isVisible: false,
}

Spacer.Horizontal = Horizontal
Spacer.Vertical = Vertical

export default Spacer
