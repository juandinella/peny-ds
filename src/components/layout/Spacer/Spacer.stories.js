import Spacer from './Spacer'
import { options } from './constants'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../../helpers/storybook'

const Template = getTemplate(Spacer)
const ListTemplate = getListTemplate(Spacer)

export default {
  title: 'Layout/Spacer',
  component: Spacer,
  args: {
    isVisible: true,
  },
  argTypes: {
    size: getOptionsArgTypes(options.sizes),
    mSize: getOptionsArgTypes(options.sizes),
  },
  parameters: { __sb: { fd: 'row' } },
}

export const Default = Template.bind({})

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }

export const MSizes = ListTemplate.bind({})
MSizes.args = { items: options.sizes.map((size) => ({ size })) }
