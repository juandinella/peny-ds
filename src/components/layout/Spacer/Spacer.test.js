import React from 'react'

import { cleanup, render } from '@testing-library/react'

import Spacer from './Spacer'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Layout\/Spacer$/,
})

const componentProps = {
  id: 'spacer',
}

afterEach(cleanup)

describe('<Spacer />', () => {
  test('should render spacer correctly', () => {
    const { queryByTestId } = render(<Spacer {...componentProps} />)

    const spacer = queryByTestId(componentProps.id)

    expect(spacer).toBeTruthy()
  })

  test('should render horizontal spacer correctly', () => {
    const { queryByTestId } = render(<Spacer.Horizontal {...componentProps} />)

    const spacer = queryByTestId(componentProps.id)

    expect(spacer).toBeTruthy()
  })

  test('should render vertical spacer correctly', () => {
    const { queryByTestId } = render(<Spacer.Vertical {...componentProps} />)

    const spacer = queryByTestId(componentProps.id)

    expect(spacer).toBeTruthy()
  })
})
