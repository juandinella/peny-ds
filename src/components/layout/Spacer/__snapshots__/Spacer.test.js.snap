// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Spacer Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-1hghm4b-0 ijeove"
    size="4"
  />
</div>
`;

exports[`Storyshots Layout/Spacer M Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-1hghm4b-0 iRleBS"
    size="0"
  />
  <div
    className="sc-1hghm4b-0 ijeove"
    size="4"
  />
  <div
    className="sc-1hghm4b-0 chFvui"
    size="8"
  />
  <div
    className="sc-1hghm4b-0 dpayLK"
    size="12"
  />
  <div
    className="sc-1hghm4b-0 hlBhmi"
    size="16"
  />
  <div
    className="sc-1hghm4b-0 jlFlyy"
    size="24"
  />
  <div
    className="sc-1hghm4b-0 gZEXkq"
    size="32"
  />
  <div
    className="sc-1hghm4b-0 qzyLK"
    size="48"
  />
  <div
    className="sc-1hghm4b-0 eUhKfK"
    size="56"
  />
  <div
    className="sc-1hghm4b-0 gUlOsa"
    size="64"
  />
</div>
`;

exports[`Storyshots Layout/Spacer Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-1hghm4b-0 iRleBS"
    size="0"
  />
  <div
    className="sc-1hghm4b-0 ijeove"
    size="4"
  />
  <div
    className="sc-1hghm4b-0 chFvui"
    size="8"
  />
  <div
    className="sc-1hghm4b-0 dpayLK"
    size="12"
  />
  <div
    className="sc-1hghm4b-0 hlBhmi"
    size="16"
  />
  <div
    className="sc-1hghm4b-0 jlFlyy"
    size="24"
  />
  <div
    className="sc-1hghm4b-0 gZEXkq"
    size="32"
  />
  <div
    className="sc-1hghm4b-0 qzyLK"
    size="48"
  />
  <div
    className="sc-1hghm4b-0 eUhKfK"
    size="56"
  />
  <div
    className="sc-1hghm4b-0 gUlOsa"
    size="64"
  />
</div>
`;
